src/drivers/device-events/index.js

```diff
@@ -0,0 +1,89 @@
+import xs from 'xstream';
+import {tap, omit, merge, identity} from 'ramda';
+
+
+const deviceEvents = [];
+
+const shakeGesture = () => {
+
+	if (!window.DeviceMotionEvent) {
+		return;
+	}
+
+
+
+	let last = null;
+
+	const threshold = 2;
+	const timeout = 1000;
+	let lastTime = new Date();
+
+	var query = window.matchMedia("(orientation:landscape)");
+	const coord = query.matches ? "x" : "y";
+
+	const deviceMotionHandler = eventData => {
+		let info = null;
+		var current = eventData.accelerationIncludingGravity;
+		var currentTime;
+		var timeDifference;
+
+		if (last === null) {
+			last = current.y;
+			return
+		}
+
+		const delta = Math.abs(last - current[coord]);
+
+		if (delta > threshold) {
+			currentTime = new Date();
+			timeDifference = currentTime.getTime() - lastTime.getTime();
+
+			if (timeDifference > timeout) {
+				deviceEvents.push({type: 'vertical-shake'});
+				lastTime = new Date();
+			}
+		}
+
+		last = current[coord];
+
+
+	};
+	window.addEventListener('devicemotion', deviceMotionHandler, false)
+};
+
+let streamInterval = 0;
+
+const deviceEventsProducer = {
```

Comment:
Sorry, but I do not understand this part. Why you need this producer?
You can create a stream from the devicemotion event and then you can map on it and do any calculation.
If the browser trigger too oftern you can compose your stream with a debounce

You can found the fromEvent and the debounce in the  XStream extra folder
https://github.com/staltz/xstream/blob/master/src/extra